eeg_flow/tasks/epochs_evoked.py

```diff
@@ -0,0 +1,434 @@
+
+    return raw, events, event_id, metadata
+
+def create_metadata(events, events_id, raw, row_events):
+    """
+    metadata for each epoch shall include events from the range: [0.0, 1.5] s,
+    i.e. starting with stimulus onset and expanding beyond the end of the epoch
+    """
+    metadata_tmin, metadata_tmax = 0.0, 0.999
+
+    """
+    auto-create metadata
+    this also returns a new events array and an event_id dictionary.
+    we'll see later why this is important
+    """
+
+    metadata, events, event_id = make_metadata(
+        events=events,
+        event_id=events_id,
+        tmin=metadata_tmin,
+        tmax=metadata_tmax,
+        sfreq=raw.info["sfreq"],
+        row_events=row_events,
+    )
+
+    conditions = [
+        metadata["event_name"].eq("target") & pd.notna(metadata["response"]),
+        metadata["event_name"].eq("target") & pd.isna(metadata["response"]),
+        metadata["event_name"].eq("standard") & pd.notna(metadata["response"]),
+        metadata["event_name"].eq("standard") & pd.isna(metadata["response"]),
+        metadata["event_name"].eq("novel") & pd.notna(metadata["response"]),
+        metadata["event_name"].eq("novel") & pd.isna(metadata["response"]),
+    ]
+    choices = [
+        "Hits",
+        "Misses",
+        "FalseAlarms",
+        "CorrectRejections",
+        "FalseAlarms",
+        "CorrectRejections",
+    ]
+
+    metadata["response_type"] = np.select(conditions, choices, default=0)
+    metadata["response_type"].value_counts()
+    
+    ########
+    metadata.response_correct = False
+    metadata.loc[
+        (metadata["response_type"] == "CorrectRejections"), "response_correct"
+    ] = True
+    metadata.loc[(metadata["response_type"] == "Hits"), "response_correct"] = True
+
+    metadata.loc[
+        (metadata["response_type"] == "FalseAlarms"), "response_correct"
+    ] = False
+    metadata.loc[(metadata["response_type"] == "Misses"), "response_correct"] = False
+
+
+    return metadata, events, event_id
+
+def get_SDT_outcomes(metadata):
+    
+    num_Hits = len(metadata[metadata["response_type"] == "Hits"])
+    num_CorrectRejections = len(
+        metadata[metadata["response_type"] == "CorrectRejections"]
+    )
+    num_Misses = len(metadata[metadata["response_type"] == "Misses"])
+    num_FalseAlarms = len(metadata[metadata["response_type"] == "FalseAlarms"])
+
+    return(num_Hits, num_CorrectRejections, num_Misses, num_FalseAlarms)
+
+def plot_RT(hits, fname_stem, derivatives_subfolder, response_mean, response_std):
+    # visualize response times of TP
+    ax_rt = hits["response"].plot.hist(
+        bins=100,
+        title=f"Response Times of TPs\nmean:{str(response_mean)} ({str(response_std)})",
+    )
+
+    fname_rt_plot = derivatives_subfolder / "plots" / (fname_stem + "_step9_RT.svg")
+    ax_rt.figure.suptitle(fname_stem, fontsize=16, y=1)
+    ax_rt.figure.savefig(fname_rt_plot, transparent=True)
+    ax_rt
```

Comment:
Why this last line `ax_rt` by iteself? It does not feel right inside a function, it would feel OK inside a jupyter-cell.

Reply:
probably a remnant of the jupyterlab version!